### DISCUSSION TITLE
Sorting (DESC or ASC) by using a numeric attribute value

### DISCUSSION THREAD

**Discussion by TurkaljGoran** [saleor/saleor#10490] [Q&A]
Hello,

I'm currently trying to sort products in ascending or descending order based on numeric attribute values but ran into a problem since the sorting is done on string values. 

So if a numeric atribute has values 1,3,11 and I try to sort products, GraphQL will return 1,11,3 because its sorting based on strings, not 1,3,11 if it were sorting numbers.
 
Is there currently any option to 'tell' Saleor to use actual number values in graphql if an attribute is numeric, since we get strings no matter what type of attribute we use, or does this require customizing the graphql functionality?

Any info is appreciated! 

Cheers!

**Comment by peelar** [comment]
Hello @TurkaljGoran 👋 

Thank you for pointing this out. Could you provide a GraphQL query for the [demo.saleor.io](https://demo.saleor.io/) environment with the replication of this behavior?

**Comment by RZ-Jyothsna** [comment]
Hi! How did you sort products by attribute value?

**Comment by linhvovan29546** [comment]
Hi @peelar 
I also have the same problem. I tried to reproduce the query for https://learn.saleor.cloud/graphql/ environment as example but can't. Because the demo products don't have any attribute type numeric
### Steps to reproduce:
 I have an attribute `vintage` with `Numeric` type, and when I try to sort products by that attribute:
- Actual result:
GraphQL will return 2,2012,3 
- Expected result:
GraphQL will return 2,3,2012
### My GraphQL:
<img width="904" alt="image" src="https://github.com/saleor/saleor/assets/84692183/86af1f9b-7e25-4fb4-b448-313f6e02d1d4">
<img width="940" alt="image" src="https://github.com/saleor/saleor/assets/84692183/206b450f-de11-4bab-8707-84dc76154e58">
